### ISSUE TITLE
Test/Prod support

### ISSUE THREAD

**Issue by scphantm** [rafael-brandao/scalabuff-gradle-plugin#2]
Im going thru the code, it looks like this doesn't support test proto's alongside main proto's.  Am i correct?

**Comment by rafael-brandao** [comment]
Yes, the plugin adds the `scalabuff.generated-sources` dir to the main scala classpath, but i'm thinking about removing this step and leave to the client the responsibility of choose which compile task to attach it to (scalaCompile or scalaTestCompile or custom ScalaCompile...)

**Comment by scphantm** [comment]
yea, it would be very helpful to separate them.  The protobuf plugin im using right now wants a src/main/proto and a src/test/proto folder.  both compile to the build/generated-sources/test and build/generated-sources/main folders.  im sure i can make it work with other source sets as well.  The type of project i am doing, there really is no other way to test some of the functions without custom proto's that have no purpose in the main source set.  

This was actually the deciding factor in stopping to use this plugin.  It straight up won't work for me without it.  Im using this protobuf plugin if your want to take a look 

classpath 'ws.antonov.gradle.plugins:gradle-plugin-protobuf:0.9.1'

**Comment by rafael-brandao** [comment]
I'm working on another project right now, but I'll definitely implement this sometime.
I've used antonov plugin's before, it's very good, except for not generating idiomatic scala code.
